### DISCUSSION TITLE
updating entries / fuzzy search

### DISCUSSION THREAD

**Discussion by AndiLeni** [loupe-php/loupe#37] [Q&A] [answered]
Hello, 

Is it possible to update a search entry in the database? 

And is fuzzy searching possible?

Kind regards

**Comment by Toflar** [comment, marked as answer]
Yes, `addDocument()` will replace the existing document in case the identifier matches.
Fuzzy search: Depends what your definition of "fuzzy search" is. It is typo tolerant and all of the configuration options are documented.

**Reply by AndiLeni** [reply]
Thanks for the quick answer! This is what I was looking for. 
May I also ask you, whether caching search results is possible or planned for future releases?

**Reply by Toflar** [reply]
At this stage, I don't think it should be part of Loupe. It's very easy to just wrap your `->search()` call in your own cache method so you can always do that yourself. But I'm not ruling it out completely - somebody has to do the work and be convincing :)

**Reply by AndiLeni** [reply]
Sounds reasonable :)
~~When I need some statistics, f.e. number of indexed entries, querying the SQLite database directly is currently the only option?~~
Found it, countDocuments() should do what I need :)

**Reply by Toflar** [reply]
Yeah, the other statistics are probably not needed in production. For debugging purposes, you would always inspect the database using an SQLite capable database client anyway :)

**Reply by Toflar** [reply]
By the way: https://github.com/loupe-php/loupe/pull/47 should make caching easer ;)
